src/transport/mqtt/config/mod.rs

```diff
@@ -365,21 +365,26 @@ impl MqttConfig {
 
         mqtt_opts.set_transport(transport);
 
+        // Set the clean_session since this is the first connection.
+        mqtt_opts.set_clean_session(true);
```

Comment:
Also when the reconnect() is called, clean session will be set to true (since). Is this the behavior we want?

Reply:
Now we reset the clean_session upon successful connection or publish